pkg/virt-controller/watch/vm.go

```diff
@@ -2615,9 +2616,32 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 	// hotplugged volumes and interfaces
 	if c.needsSync(key) && syncErr == nil {
 		vmCopy := vm.DeepCopy()
-		if c.clusterConfig.HotplugNetworkInterfacesEnabled() {
-			if err = c.handleDynamicIfaceRequestOnVMI(vmCopy, vmi); err != nil {
-				syncErr = &syncErrorImpl{fmt.Errorf("Error encountered when trying to apply interface request on vmi: %v", err), HotPlugNetworkInterfaceErrorReason}
+		if c.clusterConfig.HotplugNetworkInterfacesEnabled() &&
+			vmi != nil && vmi.DeletionTimestamp == nil {
+			vmiCopy := vmi.DeepCopy()
```

Comment:
I think this can be avoided, but we can do it as a follow up.

Reply:
I will post a PR for reducing object cloning.